vidarr-pluginapi/src/test/java/ca/on/oicr/gsi/vidarr/TaggedUnionInputTypeTest.java

```diff
@@ -0,0 +1,211 @@
+package ca.on.oicr.gsi.vidarr;
+
+import static ca.on.oicr.gsi.vidarr.VeryLongString.VERY_LONG;
+
+import ca.on.oicr.gsi.Pair;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.function.ThrowingRunnable;
+
+public class TaggedUnionInputTypeTest extends InputTypeTest {
+  String taggedUnionOneFieldStringFormat = "{\"is\":\"tagged-union\",\"options\":{\"%s\":\"%s\"}}";
+
+  @Override
+  public void testSerialize() {
+    for (Map.Entry<InputType, String> type : primitiveTypes.entrySet()) {
+      serializeTester(
+          String.format(taggedUnionOneFieldStringFormat, "field", type.getValue()),
+          InputType.taggedUnion(new Pair<>("field", type.getKey())));
+    }
+  }
+
+  @Test
+  public void testSerializeMany() {
+    List<Pair<String, InputType>> varargs = new LinkedList<>();
+    ObjectNode root = MAPPER.createObjectNode(), options = MAPPER.createObjectNode();
+    root.set("is", MAPPER.convertValue("tagged-union", JsonNode.class));
```

Comment:
Oh, BTW, there's an overloaded `put` method that automatically serialises the primitive JSON types.

Reply:
"'put(java.lang.String, com.fasterxml.jackson.databind.JsonNode)' is deprecated" ?

Reply:
`put(String, String)` should be okay.

Reply:
Done